bundles/SystemInfoBundle/public/js/startup.js

```diff
@@ -0,0 +1,99 @@
+pimcore.registerNS('pimcore.bundle.system_info.startup');
+
+pimcore.bundle.system_info.startup = Class.create({
+    user: null,
+    toolbar: null,
+    perspectiveConfig: null,
+
+    initialize: function(){
+        document.addEventListener(pimcore.events.preMenuBuild, this.preMenuBuild.bind(this));
+    },
+
+    preMenuBuild: function (event) {
+        const menu = event.detail.menu;
+        this.user = pimcore.globalmanager.get('user');
+        this.toolbar = pimcore.globalmanager.get('layout_toolbar');
+        this.perspectiveConfig = pimcore.globalmanager.get('perspective');
+        const systemInfoMenuItems = this.getSystemInfoMenu();
+
+        const filteredMenu = menu.extras.items.filter(function (item) {
+            return item.itemId === 'pimcore_menu_extras_system_info';
+        });
+
+        if (filteredMenu.length > 0) {
+            const systemInfoMenu = filteredMenu.shift();
+            systemInfoMenuItems.map(function(item) {
+                systemInfoMenu.menu.items.push(item);
+            });
+        } else {
+            menu.extras.items.push({
+                text: t("system_infos_and_tools"),
+                iconCls: "pimcore_nav_icon_info",
+                hideOnClick: false,
+                itemId: 'pimcore_menu_extras_system_info',
```

Comment:
With this approach here one may have to "sync" it with any changes in the core.

To overcome that, please see similar changes like here
https://github.com/pimcore/pimcore/pull/13967/files#diff-0bd635bc68ae1ed97161f850140cef5707f9be71d5dd982c1f978c6aab2255c7R321

In short, it always adds `pimcore_menu_extras_system_info` in core, (that's why it's moved outside the IF block) and by the changes https://github.com/pimcore/pimcore/pull/14020 would recursively clean/skip the submenus with no items.